arches/app/models/graph.py

```diff
@@ -367,6 +367,7 @@ def add_node(self, node, nodegroups=None):
                 node.nodegroup = self.get_or_create_nodegroup(
                     nodegroupid=node.nodegroup_id
                 )
+                node.nodegroup.grouping_node_id = node.nodegroup_id
```

Comment:
should this instead be part of `get_or_create_nodegroup`?

Reply:
I like it 🧠